CMakeLists.txt

```diff
@@ -20,7 +20,10 @@ endif()
 
 enable_testing()
 
-add_definitions(-std=c++11)
+add_definitions(
+  -std=c++11
+  -DBOOST_FILESYSTEM_NO_DEPRECATED
```

Comment:
For details regarding the added `-DBOOST_FILESYSTEM_NO_DEPRECATED` definition see [Boost Filesystem Deprecated Features](https://www.boost.org/doc/libs/1_81_0/libs/filesystem/doc/deprecated.html)